internal/cephfs/core/metadata.go

```diff
@@ -29,34 +32,82 @@ const (
 // GetMetadata gets custom metadata on the subvolume in a volume based on key
 // name.
 func (s *subVolumeClient) GetMetadata(key string) (string, error) {
+	notSupErr := errors.New("operation GetMetadata is not supported")
+	if clusterAdditionalInfo[s.clusterID].subVolMetadataState == unsupported {
```

Comment:
this is a very long if-statement, and it repeats itself often. Much cleaner to have a helper function like
```go
func (s *subVolumeClient) supportsSubVolMetadata() bool {}
```